operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -291,10 +308,35 @@ func (r *TenantReconciler) checkValidWorkspaces(ctx context.Context, tn *crownla
 			tnOpinternalErrors.WithLabelValues("tenant", "workspace-not-exist").Inc()
 		} else {
 			tenantExistingWorkspaces = append(tenantExistingWorkspaces, tnWs)
-			workspaces = append(workspaces, ws)
+			if tnWs.Role == crownlabsv1alpha2.Candidate {
+				if !slices.Contains(possibleEnrollFeatures, string(ws.Spec.AutoEnroll)) {
```

Comment:
Would make a helper function for this, saw you also did this on the webhook, just not to also have the duplicated `possibleEnrollFeatures` and such...

Reply:
Ok. Since webhook and tenant controller are in different packages, I'll insert the function in the `utils` package